UM/View/GL/ShaderProgram.py

```diff
@@ -348,18 +348,25 @@ def removeAttributeBinding(self, key: str) -> None:
         del self._attribute_bindings[key]
 
     def _matrixToQMatrix4x4(self, m):
-        return QMatrix4x4(m.getData().flatten())
+        return QMatrix4x4(m.getFlatData())
 
     def _setUniformValueDirect(self, uniform: int, value: Union[Vector, Matrix, Color, List[float], List[List[float]], float, int]) -> None:
-        if type(value) is Vector:
+        if type(value) is Matrix:
```

Comment:
It's not _completely_ clear (to me) we set Vectors more often than Matrices (at least with the current state of our code). Take a look at all the `vec2`, `vec3` and `vec4` in our shaders and compare that to the occurrences of `mat4`. Then again the camera is the thing that _should_ change the most, if we set the others more often then a lot of those are probably spurious changes.

Reply:
Well, I just ran this through a profiler and that is what I found. But the effect of changing the order is probably quite small.